### PR TITLE
Handle exports similar to imports in the parser.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
Currently we don't handle exports in any way and exposing those to the storyboard creation would mean we could handle React components that aren't straight exposed with the `export` modifier.

**Fix:**
Have the parser process and handle exports similarly to exports conglomerating the various ways something can be exported.

**Limitations:**
This does not currently handle things like inline anonymous components that are exposed with `export default` and the `export default Thing` style which can expose an existing constant or variable. Thought it better to checkpoint this work here for now as it covers the overall flow.

**Commit Details:**
- Created `ExportsDetail` along with associated types to represent
  the detail of what is exported, attempting to pull together the
  individual representations of the exports.
- Added `exportsDetail` to `ParseSuccess`.
- Implemented `isDefaultExport`.
- Removed `TopLevelElementAndCodeContext` as it wasn't really used.
- Printer now applies modifiers to elements based on the export detail.
- Printer adds in an export expression as appropriate if values are
  exported that way.
- Parser now also processes export declarations and includes the values
  from those along with the modifiers from JSX components parsed.
- Changed some import sorting to use our comparison code instead of
  a hand rolled function.

### CODE REVIEW

File: editor/src/core/workers/parser-printer/parser-printer-test-utils.ts [hunk 1]
```diff
@@ -484,6 +493,36 @@ export function topLevelElementArbitrary(): Arbitrary<TopLevelElement> {
...
+    exportDetailModifierArbitrary(),
+  )
+}
+
+export function exportsDetailArbitrary(possibleNames: Array<string>): Arbitrary<ExportsDetail> {
+  if (possibleNames.length === 0) {
+    return FastCheck.constant(EmptyExportsDetail)
+  } else {
+    return FastCheck.tuple(
+      FastCheck.constant(null), //FastCheck.option(exportDetailNamedArbitrary(possibleNames)),
```
Comment (Rheeseyb) [inline editor/src/core/workers/parser-printer/parser-printer-test-utils.ts:518]: What's with the commented code here?

Comment (seanparsons) [reply editor/src/core/workers/parser-printer/parser-printer-test-utils.ts:518]: Ah yes, this is something I need to follow up on, it relates to `export default Thing = (props) => {...}` being invalid.

File: editor/src/core/workers/parser-printer/parser-printer.spec.ts [hunk 2]
```diff
@@ -1907,44 +2010,14 @@ export var whatever = (props) => <View data-uid={'aaa'}>
     const exported = utopiaJSXComponent('whatever', true, defaultPropsParam, ['color'], view, null)
     const imports = addImport('cake', null, [importAlias('cake')], null, sampleImportsForTests)
     const expectedResult = clearParseResultUniqueIDs(
-      parseSuccess(imports, [exported], expect.objectContaining({}), null, null),
-    )
-    expect(actualResult).toEqual(expectedResult)
-  })
-  it('parses the code when it is a var', () => {
```
Comment (Rheeseyb) [inline editor/src/core/workers/parser-printer/parser-printer.spec.ts:1914]: What happened to this test?

File: website/package-lock.json [hunk 3]
```diff
@@ -15496,9 +15496,9 @@
       "integrity": "sha1-hnrHTjhkGHsdPUfZlqeOxciDB3c="
     },
     "typescript": {
-      "version": "4.0.2",
-      "resolved": "https://registry.npmjs.org/typescript/-/typescript-4.0.2.tgz",
-      "integrity": "sha512-e4ERvRV2wb+rRZ/IQeb3jm2VxBsirQLpQhdxplZ2MEzGvDkkMmPglecnNDfSUBivMjP93vRbngYYDQqQ/78bcQ==",
+      "version": "4.0.5",
```
Comment (Rheeseyb) [inline website/package-lock.json:15499]: Oops, I must have not committed this before :grimacing:

File: editor/src/core/workers/parser-printer/parser-printer.spec.ts (hunk 2 above)
Comment (seanparsons) [reply editor/src/core/workers/parser-printer/parser-printer.spec.ts:1914]: It's a duplicate:
https://github.com/concrete-utopia/utopia/blob/a9c7edc3ffe8b8a7e0b04d5a7dd6dea9d97ed1a9/editor/src/core/workers/parser-printer/parser-printer.spec.ts#L1914
https://github.com/concrete-utopia/utopia/blob/a9c7edc3ffe8b8a7e0b04d5a7dd6dea9d97ed1a9/editor/src/core/workers/parser-printer/parser-printer.spec.ts#L1951

Comment (Rheeseyb) [reply editor/src/core/workers/parser-printer/parser-printer.spec.ts:1914]: Ha, seems like that was my fault when removing old dead code!

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/core/workers/parser-printer/parser-printer.spec.ts (hunk 2 above)
Comment (seanparsons) [reply editor/src/core/workers/parser-printer/parser-printer.spec.ts:1914]: I totally went "Wait, I fixed this test..." when changing the test.

File: editor/src/core/workers/parser-printer/parser-printer-arbitrary-elements.spec.ts [hunk 4]
```diff
@@ -200,6 +206,7 @@ export var whatever = props => (
       expect.objectContaining({}),
       null,
       null,
+      addModifierExportToDetail(EmptyExportsDetail, 'whatever'),
```
Comment (balazsbajorics) [inline editor/src/core/workers/parser-printer/parser-printer-arbitrary-elements.spec.ts:209]: does this mean `whatever` is exported like so: `export var whatever = (props)...` ? or a separate line exporting it?

Comment (seanparsons) [reply editor/src/core/workers/parser-printer/parser-printer-arbitrary-elements.spec.ts:209]: The former.

**Review by balazsbajorics — APPROVED** [review]

**Pull request merged**